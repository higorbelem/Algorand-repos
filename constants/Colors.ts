const tintColorDark = '#fff';

export const Colors = {
  text: '#FFFFFF',
  background: '#191C21',
  tint: tintColorDark,
  icon: '#9BA1A6',
  tabIconDefault: '#9BA1A6',
  tabIconSelected: tintColorDark,
  main: '#2D2DF1',
  green: '#238636'
};
