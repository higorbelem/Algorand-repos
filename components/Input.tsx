import { StyleSheet, TextInput, TextInputProps, View } from "react-native"

import { Colors } from "@/constants/Colors";

export const Input = (props: TextInputProps) => {
    return(
        <View style={styles.container}>
            <TextInput {...props} style={styles.input} placeholderTextColor={Colors.text}/>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        borderColor: Colors.text,
        borderWidth: 1,
        height: 40,
        borderRadius: 20,
        paddingHorizontal: 15
    },
    input: {
        width: '100%',
        height: '100%',
        color: Colors.text
    },
});