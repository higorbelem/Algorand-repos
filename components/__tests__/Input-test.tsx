import * as React from 'react';
import { render } from '@testing-library/react-native';

import { Colors } from '@/constants/Colors';

import { Input } from '../Input';

describe('Input component', () => {
  it('Renders correctly', () => {
    const tree = render(<Input />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Shows correct placeholder', async () => {
    const tree = render(<Input placeholder='placeholder'/>);

    expect(await tree.findByPlaceholderText('placeholder')).toBeTruthy();
  });
  
    it('Has correct color', async () => {
      const tree = render(<Input placeholder='placeholder'/>);
      
      const component = await tree.findByTestId('input-component');
  
      expect(component).toHaveStyle({color: Colors.text});
    });
});