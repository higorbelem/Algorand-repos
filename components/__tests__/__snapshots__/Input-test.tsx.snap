// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Renders correctly 1`] = `
<View
  style={
    {
      "borderColor": "#FFFFFF",
      "borderRadius": 20,
      "borderWidth": 1,
      "height": 40,
      "paddingHorizontal": 15,
    }
  }
>
  <TextInput
    placeholderTextColor="#FFFFFF"
    style={
      {
        "color": "#FFFFFF",
        "height": "100%",
        "width": "100%",
      }
    }
    testID="input-component"
  />
</View>
`;
